script.js

```diff
@@ -1,110 +1,175 @@
+  episodeCard.appendChild(title);
+  episodeCard.appendChild(dataContainer);
 
   return episodeCard;
 }
 
-function setupSearch() {
-  const searchInput = document.getElementById("search-input");
-  searchInput.addEventListener("input", function () {
-    const searchTerm = searchInput.value.trim().toLowerCase();
-    filterEpisodes(searchTerm);
+// We're making a menu of episode numbers so people can pick a specific episode to watch.
+function populateEpisodeNumberFilter(episodeList) {
+  const episodeNumberFilterElement = document.getElementById('episode-number-filter');
+
+  // We're clearing out any old options from the menu.
+  episodeNumberFilterElement.innerHTML = "";
+
+  // We're making an option to show all episodes in case people want to see them all.
+  const defaultOption = document.createElement('option');
+  defaultOption.value = 'all';
+  defaultOption.textContent = 'Show All Episodes';
+  episodeNumberFilterElement.appendChild(defaultOption);
+
+  // We're adding each episode number to the menu so people can pick one.
+  episodeList.forEach(episode => {
+      const optionElement = document.createElement('option');
+      optionElement.value = `S${episode.season.toString().padStart(2, '0')}E${episode.number.toString().padStart(2, '0')}`;
+      optionElement.textContent = `${optionElement.value} - ${episode.name}`;
+      episodeNumberFilterElement.appendChild(optionElement);
   });
-}
-
-function filterEpisodes(searchTerm) {
-  const filteredEpisodes = allEpisodes.filter(
-    (episode) =>
-      episode.name.toLowerCase().includes(searchTerm) ||
-      episode.summary.toLowerCase().includes(searchTerm)
-  );
 
-  makePageForEpisodes(filteredEpisodes);
+  // When someone picks an episode from the menu, we want to show that episode and hide the others.
+  episodeNumberFilterElement.addEventListener('change', function () {
+      const selectedEpisode = this.value;
+      filterEpisodesByEpisodeNumber(selectedEpisode, episodeList);
+  });
 }
 
-function updateEpisodeListCounter(episodeList) {
-  const episodesDisplayAmount = document.querySelector(
-    ".episodes-display-amount"
-  );
-  episodesDisplayAmount.textContent = `Displaying ${episodeList.length} out of ${allEpisodes.length} episodes`;
+// When someone picks an episode from the menu, we want to show that episode and hide the others.
+function filterEpisodesByEpisodeNumber(selectedEpisode, episodeList) {
+  const rootElem = document.getElementById("root");
+
+  episodeList.forEach(episode => {
+      const episodeCard = rootElem.querySelector(`.episode-card__title`);
+      if (episodeCard.textContent.includes(selectedEpisode) || selectedEpisode === 'all') {
+          // If the episode matches the one selected or if all episodes are selected, we show it.
+          episodeCard.parentElement.style.display = 'block';
+      } else {
+          // If it doesn't match, we hide it.
+          episodeCard.parentElement.style.display = 'none';
+      }
+  });
```

Comment:
When I select an episode in the dropdown menu, it doesn't display the correct episode.
I suspect the problem lies in the logic behind the variable selectedEpisode, which sends the select element's value and searches the entire textContent, or it could be related to the .includes method.